protocol/lib/vault/utils.go

```diff
@@ -0,0 +1,51 @@
+package vault
+
+import (
+	"math/big"
+
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	pricestypes "github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+)
+
+// SkewAntiderivativePpm returns the antiderivative of skew given a vault's skew
+// factor and leverage.
+// skew_antiderivative_ppm = skew_factor * leverage^2 + skew_factor^2 * leverage^3 / 3
+func SkewAntiderivativePpm(
+	skewFactorPpm uint32,
+	leveragePpm *big.Int,
+) *big.Int {
+	bigSkewFactorPpm := new(big.Int).SetUint64(uint64(skewFactorPpm))
+	bigOneTrillion := lib.BigIntOneTrillion()
+
+	// a = skew_factor * leverage^2.
+	a := new(big.Int).Mul(leveragePpm, leveragePpm)
+	a.Mul(a, bigSkewFactorPpm)
+
+	// b = skew_factor^2 * leverage^3 / 3.
+	b := new(big.Int).Set(a)
+	b.Mul(b, leveragePpm)
+	b.Mul(b, bigSkewFactorPpm)
+	b = lib.BigDivCeil(b, big.NewInt(3))
+
+	// normalize a and b.
```

Comment:
nit: Maybe a bit more of a comment on the normalization (ppm * ppm * ppm / 1,000,000,000,000) = (ppm)?